packages/runtime/src/runtime/hooks/useEnsembleAction.tsx

```diff
@@ -320,6 +341,37 @@ export const useInvokeAPI: EnsembleActionHook<InvokeAPIAction> = (action) => {
 
     const fireRequest = async (): Promise<void> => {
       setIsLoading(true);
+      // First check if useMockResponse is enabled and if a mockResponse exists
+      if (
+        api.mockResponse &&
+        isUsingMockResponse(appContext?.application?.id)
+      ) {
+        // Fetch the mock response and set the api data
+        const mockResponse = mock({ response: mockResponses[api.name] });
+        setData(api.name, mockResponse);
+
+        // Next, run the appropriate actions
+        if (action?.id) {
+          setData(action.id, mockResponse);
+        }
+
+        if (mockResponse.isSuccess) {
+          onAPIResponseAction?.callback({ ...context, response: mockResponse });
+          onInvokeAPIResponseAction?.callback({
+            ...context,
+            response: mockResponse,
+          });
+        } else {
+          onAPIErrorAction?.callback({ ...context, error: mockResponse });
+          onInvokeAPIErrorAction?.callback({ ...context, error: mockResponse });
+        }
+
+        setIsComplete(true);
+        setIsLoading(false);
+        return;
```

Comment:
Rather than repeating this logic can we just move the `mock` into the datafetcher?